### ISSUE TITLE
deploy dependabot submodule version update github action across repos

### ISSUE THREAD

**Issue by kbroch-rivosinc** [riscv-admin/docs-sig#63]
Discussed in doc-sig meeting today.  As @edolnx brought up could be applied [across org's](https://docs.github.com/en/code-security/securing-your-organization/enabling-security-features-in-your-organization/applying-the-github-recommended-security-configuration-in-your-organization#applying-the-github-recommended-security-configuration-to-all-repositories-in-your-organization)

Currently here's the repos using dependabot:

```
❯ fd -H dependabot.yml
riscv/sig-functional-safety-whitepaper/.github/dependabot.yml
riscv/riscv-isa-manual/.github/dependabot.yml
riscv/docs-spec-template/.github/dependabot.yml
riscv-admin/policies/.github/dependabot.yml
```

Here's number of repos that could benefit from it:

```
❯ for dir in $(ls -1 | tr "\n" " "); do echo "$dir: $(find "$dir" -type f -name .gitmodules | wc -l)"; done
riscv:       41
riscv-admin:        4
riscv-collab:        2
riscv-non-isa:       30
riscv-software-src:        3
```

NOTE: riscv-android-src exclude

**Comment by kbroch-rivosinc** [comment]
not surprising but means definitely don't deploy this to repo's w/o submodules:
https://github.com/riscv-admin/docs-sig/network/updates/927327612
<img width="1452" alt="image" src="https://github.com/user-attachments/assets/33dd042f-8bde-4185-9928-688a6808f33b">

**Comment by kbroch-rivosinc** [comment]
Yes, I will selectively apply based on whether repo has a `.gitmodules` file.

**Comment by kbroch-rivosinc** [comment]
Deployment was a bit more involved as I needed to fork the projects as I am not a maintainer on almost all of them.   I was able to use `gh` to automate that.  NOTE: gh said I had to fork to an "org" not an "user" but luckily, but that's wasn't hard to just create an org.

After that I just had to define another "downstream" remote of the forks and push to that. 

For the creation of the PR's I just clicked on the "Create pull request links" and created them manually in the web browser.

**Comment by wmat** [comment]
Awesome, what's the run frequency of dependabot?  Does it monitor the
submodule repo for changes?

**Comment by kbroch-rivosinc** [comment]
> Awesome, what's the run frequency of dependabot? Does it monitor the submodule repo for changes?

Run frequency is daily: https://github.com/riscv/docs-spec-template/blob/main/.github/dependabot.yml#L8

Yes, once dependabot.yml file is in place and dependabot settings are enabled, it will run, and generate a PR if the existing submodule ref(s) are behind.  ex: https://github.com/riscv/sig-functional-safety-whitepaper/pull/2

Do you think the owners of the orgs should just [enable it](https://docs.github.com/en/code-security/securing-your-organization/enabling-security-features-in-your-organization/applying-the-github-recommended-security-configuration-in-your-organization#applying-the-github-recommended-security-configuration-to-all-repositories-in-your-organization) for everyone?  If not, might need to announce to repo owners to do it in order to to see the change.

**Comment by wmat** [comment]
Ideally everyone should be using the latest docs-resources, however some
folks may have a dependency on old versions due to constraints on their
content, such as tables with lots of columns and small fonts, etc.

Can I manually trigger dependabot to update the docs-resources by
re-running the job?

**Comment by kbroch-rivosinc** [comment]
> Can I manually trigger dependabot to update the docs-resources by re-running the job?

Haven't tried that.  Let me know if it works :)

**Comment by wmat** [comment]
Yep, you can.  Goto Insights -> Dependency Graph -> Dependabot tab -> click
check for updates.

**Comment by wmat** [comment]
Can we utilize repo to add or update .github/release.yml?  Christian Herber
rightfully pointed out to me that dependabot may pollute the changelogs.  This
can be prevented, by blocking the author dependabot from the changelog
https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

I.e. add a .github/release.yml, with changelog.exclude.authors set to
dependabot.

**Comment by kbroch-rivosinc** [comment]
> Can we utilize repo to add or update .github/release.yml? Christian Herber rightfully pointed out to me that dependabot may pollute the changelogs. This can be prevented, by blocking the author dependabot from the changelog https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes I.e. add a .github/release.yml, with changelog.exclude.authors set to dependabot.

I don't see many/any projects that would be affected by this.  Seems like it can wait till it actually becomes a problem.

**Comment by wmat** [comment]
The ISA manuals would be hit by this, I think.

**Comment by kbroch-rivosinc** [comment]
> The ISA manuals would be hit by this, I think.

Here's the first [instance of this](https://github.com/riscv/riscv-isa-manual/releases/tag/riscv-isa-release-067b86b-2024-12-16).  I'll leave it to the maintainers of isa to decide if this "pollutes" the changelog.  IMO it is good to see this change occurred as it could have real impact on the resulting release.

**Comment by wmat** [comment]
I went ahead and merged the last change to docs-resources.  My thinking is
there shouldn't be that many updates to docs-resources, so the changelogs
shouldn't be hit too hard.
